services/venice-controller/src/main/java/com/linkedin/venice/ingestion/control/RealTimeTopicSwitcher.java

```diff
@@ -51,6 +54,10 @@ public RealTimeTopicSwitcher(
     this.destKafkaBootstrapServers = veniceProperties.getBoolean(ConfigKeys.SSL_TO_KAFKA, false)
         ? veniceProperties.getString(ConfigKeys.SSL_KAFKA_BOOTSTRAP_SERVERS)
         : veniceProperties.getString(ConfigKeys.KAFKA_BOOTSTRAP_SERVERS);
+    int kafkaReplicationFactor = veniceProperties.getInt(KAFKA_REPLICATION_FACTOR);
+    this.kafkaReplicationFactorForRTTopics =
+        veniceProperties.getInt(KAFKA_REPLICATION_FACTOR_RT_TOPICS, kafkaReplicationFactor);
+    this.minSyncReplicasForRTTopics = veniceProperties.getOptionalInt(KAFKA_MIN_IN_SYNC_REPLICAS_RT_TOPICS);
```

Comment:
Can we set the default values for `minSyncReplicasForRTTopics`, `minInSyncReplicas`, and `minInSyncReplicasAdminTopics` and get rid of Optional?

Reply:
It's difficult to set default for min in sync replicas since it's related to replication factor. If we set a default and only override RF then it might lead to some unintended behavior. Whereas providing the Optional.empty() will let broker set the correct default min ISR based on the RF set.